documentation/project/contributing/building.md

```diff
@@ -1,24 +1,62 @@
-### Building the application locally
+## Build & Run
 
-Once you installed the prerequisites and cloned the repository, run the following commands in your project directory:
+Once you installed the prerequisites and cloned the repository, run the following steps in your project directory:
 
-Build a docker container with the app:
+#### Step 1 : Build
+> **_NOTE:_**  If you are an macOS M1 User then please keep in mind below things
+
+> Skip the maven tests as they might not be successful
+
+> Comment out `com.spotify` plugin in `kafka-ui-api` pom file
+
+> Once your build is successful you need to create docker image manually as you'd commented `com.spotify` plugin
+
+> ```docker build --platform linux/amd64 -t provectuslabs/kafka-ui:latest --build-arg JAR_FILE=kafka-ui-api-0.0.1-SNAPSHOT.jar .```
+
+
+- Build `kafka-ui-react-app`
+Follow all the steps [kafka-ui-react-app-build-documentation](../../../kafka-ui-react-app/README.md)
```

Comment:
uuhm that'd be nice to provide a link like this here but it looks like it's a required step, but the previous one was to build the whole jar. I'd suggest naming this part like "if you need to build the frontend, go here"

Reply:
@shubhwip can we update this part a bit?:)

Reply:
Sure Ok :) I remember generate sources was still mandatory step. should i include `pnpm gen:sources` as a necessary step prior to building and running backend ?

Reply:
I'm not quite sure that it's always the case, it's present in both contract and api modules (full search for `pnpm gen:sources`). I dunno why it was a problem in your case.

Reply:
Yeah true i also saw that. Let me try again with clean project and see if this is a problem still but for now we can update with your suggested changes.